### PR TITLE
react-carousel-component - Luc-Ky Khuu

### PR DESCRIPTION
**Pull request by luc-ky-khuu**
react-carousel-component

### CODE REVIEW

File: react-carousel-component/src/carousel.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+const images = ['001', '004', '007', '025', '039'];
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      interval: setInterval(() => this.nextImage(), 3000),
```
Comment (thebearingedge) [inline react-carousel-component/src/carousel.jsx]: Starting your initial interval is best done in `componentDidMount`

https://reactjs.org/docs/react-component.html#componentdidmount

File: react-carousel-component/src/carousel.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    this.prevImage = this.prevImage.bind(this);
+    this.jumpToImage = this.jumpToImage.bind(this);
+  }
+
+  nextImage() {
+    clearInterval(this.state.interval);
+    let counter = this.state.imageIndex;
+    if (counter === images.length - 1) {
+      counter = -1;
+    }
```
Comment (thebearingedge) [inline react-carousel-component/src/carousel.jsx:28]: Once you know the correct index, you can pass it to `this.jumpToImage()` 😉

File: react-carousel-component/src/carousel.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    });
+
+  }
+
+  prevImage() {
+    clearInterval(this.state.interval);
+    let counter = this.state.imageIndex;
+    if (counter === 0) {
+      counter = images.length;
+    }
```
Comment (thebearingedge) [inline react-carousel-component/src/carousel.jsx]: Once you know the correct index, you can pass it to `this.jumpToImage()` 😉

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by luc-ky-khuu** [comment]
react-carousel-component-update

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**